### PR TITLE
remove incorrect and unnecessary overrides from BooleanVectorValueMatcher

### PR DESCRIPTION
**Pull request by clintropolis**
`BooleanVectorValueMatcher.getMaxVectorSize` is incorrect (returning current vector size instead of max vector size), and the implementations provided by `BaseVectorValueMatcher` should be adequate, so the overrides have been removed.

The added test case would fail with errors of the form: 
```
2020-06-09T02:45:44,236 ERROR [main] org.apache.druid.query.groupby.epinephelinae.GroupByMergingQueryRunnerV2 - Exception with one of the sequences!
java.lang.ArrayIndexOutOfBoundsException: 2
	at org.apache.druid.query.aggregation.FilteredVectorAggregator.aggregate(FilteredVectorAggregator.java:110) ~[classes/:?]
	at org.apache.druid.query.aggregation.AggregatorAdapters.aggregateVector(AggregatorAdapters.java:200) ~[classes/:?]
...
```
prior to this patch.

### CODE REVIEW

File: processing/src/main/java/org/apache/druid/query/filter/vector/BooleanVectorValueMatcher.java [hunk 1]
```diff
@@ -38,18 +36,6 @@ public static BooleanVectorValueMatcher of(final VectorSizeInspector selector, f
     return new BooleanVectorValueMatcher(selector, matches);
   }
 
-  @Override
-  public int getCurrentVectorSize()
-  {
-    return selector.getCurrentVectorSize();
```
Comment (jihoonson) [inline processing/src/main/java/org/apache/druid/query/filter/vector/BooleanVectorValueMatcher.java:44]: nit: maybe good to have unit tests for future proof.

File: processing/src/test/java/org/apache/druid/query/filter/vector/VectorValueMatcherColumnProcessorFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    EasyMock.expect(selector.getValueCardinality()).andReturn(0).anyTimes();
+    EasyMock.replay(selector);
+
+    VectorValueMatcherFactory matcherFactory =
+        VectorValueMatcherColumnProcessorFactory.instance().makeSingleValueDimensionProcessor(selector);
+
+    Assert.assertTrue(matcherFactory instanceof SingleValueStringVectorValueMatcher);
+
+    VectorValueMatcher matcher = matcherFactory.makeMatcher("any value");
+    Assert.assertTrue(matcher instanceof BooleanVectorValueMatcher);
```
Comment (maytasm) [inline processing/src/test/java/org/apache/druid/query/filter/vector/VectorValueMatcherColumnProcessorFactoryTest.java:175]: If value cardinality == 0, shouldn't it return this thing...
     ```
 // Evaluate "lookupName" and "predicate" on every row.
      return new BaseVectorValueMatcher(selector)
      {
        final VectorMatch match = VectorMatch.wrap(new int[selector.getMaxVectorSize()]);

        @Override
        public ReadableVectorMatch match(final ReadableVectorMatch mask)
        {
          final int[] vector = selector.getRowVector();
          final int[] selection = match.getSelection();

          int numRows = 0;

          for (int i = 0; i < mask.getSelectionSize(); i++) {
            final int rowNum = mask.getSelection()[i];
            if (predicate.apply(selector.lookupName(vector[rowNum]))) {
              selection[numRows++] = rowNum;
            }
          }

          match.setSelectionSize(numRows);
          assert match.isValid(mask);
          return match;
        }
      };
```

Comment (clintropolis) [reply processing/src/test/java/org/apache/druid/query/filter/vector/VectorValueMatcherColumnProcessorFactoryTest.java:175]: With cardinality of 0, `ValueMatchers.toBooleanIfPossible` will return 'false' from the path from https://github.com/apache/druid/blob/17cf8ea8f2cc6a4fe8e2c6e29e3b9f066a6ccc2b/processing/src/main/java/org/apache/druid/query/filter/vector/SingleValueStringVectorValueMatcher.java#L62, which will make a false boolean matcher, so the current test is correct afaict.

Comment (maytasm) [reply processing/src/test/java/org/apache/druid/query/filter/vector/VectorValueMatcherColumnProcessorFactoryTest.java:175]: Ahh I see. Got it.

**Comment by clintropolis** [comment]
Thanks for the review @maytasm and @jihoonson :metal:

**Pull request merged**